diff --git a/backend/drm/backend.c b/backend/drm/backend.c
index f91492ac..938aa920 100644
--- a/backend/drm/backend.c
+++ b/backend/drm/backend.c
@@ -224,10 +224,16 @@ struct wlr_backend *wlr_drm_backend_create(struct wlr_session *session,
 		goto error_event;
 	}
 
+
+	drm->session_destroy.notify = handle_session_destroy;
+	wl_signal_add(&session->events.destroy, &drm->session_destroy);
+
 	if (drm->parent) {
 		if (!init_drm_renderer(drm, &drm->mgpu_renderer)) {
 			wlr_log(WLR_ERROR, "Failed to initialize renderer");
-			goto error_resources;
+			wl_list_remove(&drm->parent_destroy.link);
+			drm->parent = NULL;
+			return &drm->backend;
 		}
 
 		// We'll perform a multi-GPU copy for all submitted buffers, we need
@@ -237,7 +243,10 @@ struct wlr_backend *wlr_drm_backend_create(struct wlr_session *session,
 			wlr_renderer_get_dmabuf_texture_formats(renderer);
 		if (texture_formats == NULL) {
 			wlr_log(WLR_ERROR, "Failed to query renderer texture formats");
-			goto error_mgpu_renderer;
+			finish_drm_renderer(&drm->mgpu_renderer);
+			wl_list_remove(&drm->parent_destroy.link);
+			drm->parent = NULL;
+			return &drm->backend;
 		}
 
 		// Forbid implicit modifiers, because their meaning changes from one
@@ -254,15 +263,8 @@ struct wlr_backend *wlr_drm_backend_create(struct wlr_session *session,
 		}
 	}
 
-	drm->session_destroy.notify = handle_session_destroy;
-	wl_signal_add(&session->events.destroy, &drm->session_destroy);
-
 	return &drm->backend;
 
-error_mgpu_renderer:
-	finish_drm_renderer(&drm->mgpu_renderer);
-error_resources:
-	finish_drm_resources(drm);
 error_event:
 	wl_list_remove(&drm->session_active.link);
 	wl_event_source_remove(drm->drm_event);
